R/helper-functions.R

```diff
@@ -462,6 +488,43 @@ create_sparklines <- function(
   if (!is.null(markers) && as_xml_attr(markers) == "" && !is.null(type) && type %in% c("stacked", "column"))
     stop("markers only affect lines `type = NULL`, not stacked or column")
 
+  match.arg_wrapper(
+    direction,
+    c("row", "col"),
+    fn_name = "create_sparklines",
+    arg_name = "direction"
+  )
+
+  dims <- split_dims(dims, direction = direction)
+  sqref <- split_dim(sqref)
+
+  if (!length(dims) == length(sqref)) {
+    stop("dims and sqref must be equal length.")
+  }
+
+  sparkline <- Map(
+    function(dims, sqref) {
+      xml_node_create(
+        "x14:sparkline",
+        xml_children = c(
+          xml_node_create(
+            "xm:f",
+            xml_children = c(
+              paste0(shQuote(sheet, type = "sh"), "!", dims)
+            )
+          ),
+          xml_node_create(
+            "xm:sqref",
+            xml_children = c(
+              sqref
+            )
+          )
+        )
+      )
+    }, dims, sqref
+  )
+  sparklines <- paste(unlist(sparkline), collapse = "")
```

Comment:
I'm not sure I get why we need `Map()` here, but I haven't actually used it a lot. Have to run it and think about it